Schedule_app/views.py

```diff
@@ -40,22 +60,142 @@ def bookticket(response):
+
+
+def confirmation(response):
+    data = response.POST
+    reservations = []
+    for bookings in data:
+        reservations.append(bookings)
+    reservations.pop(0)
+    reservations.pop(len(reservations) - 1)
+    reservations = SeatModel.objects.filter(name__in=reservations).values_list('id', flat=True)
+    seat_list = ''
+    for seat in reservations:
+        seat_list = seat_list + '{}'.format(seat) + ','
+    seat_list = seat_list[:len(seat_list) - 1]
+    schedule_id = data['schedule_name']
+    if 'confirmation' in data:
+        BookTicket.objects.create(
+            seats=seat_list,
+            schedule_id=schedule_id,
+            user_id=response.user.id,
+            ui_confirmed=True
+        )
+        send_mail(
+            subject='Your tickets',
+            message='Here are your tickets for seats {} for this movie: {}'.format(seat_list,
+                                                                                   ScheduleMovieCinema.objects.
+                                                                                   filter(id=schedule_id)[0]),
+            from_email=settings.EMAIL_HOST_USER,
+            recipient_list=[response.user.email]
+        )
+        return HttpResponse('Booking succesfull! Check your email for your confirmation!')
+    else:
+       # user = response.user
+       # current_site = get_current_site(response)
+       # subject = 'Confirm your booking'
+       # message = render_to_string('confirmseat.html', {
+       #     'user': user,
+       #     'domain': current_site.domain,
+       #     'uid': urlsafe_base64_encode(force_bytes(user.pk)),
+       #     'token': reservation_confirmation_token.make_token(user),
+       # })
+       # to_email = user.email
+       # send_mail(
+       #     subject,
+       #     message,
+       #     settings.EMAIL_HOST_USER,
+       #     [to_email]
+       # )
+       # return HttpResponse('Check your email to confirm your seats!')
+       return HttpResponse('please go back and check te box for confimation!')
+
+
+class ConfirmBooking(View):
+
+    def get(self, request, uidb64, token, *args, **kwargs):
+        try:
+            uid = force_str(urlsafe_base64_decode(uidb64))
+            user = User.objects.get(pk=uid)
+        except (TypeError, ValueError, OverflowError, User.DoesNotExist):
+            user = None
+
+        if user is not None and reservation_confirmation_token.check_token(user, token):
+            user.bookticket.email_confirmed=True
+            messages.success(request, ('Your booking has been confirmed.'))
+            return redirect('homepage')
+        else:
+            messages.warning(request, ('The confirmation link was invalid, possibly because it has already been used.'))
+            return redirect('homepage')
+
+
+class CinemaList(APIView):
+    permission_classes = [IsAuthenticated, ]
+
+    def post(self, request, format=None):
```

Comment:
An endpoint that lists something must function under the `GET` http method. `POST` is an unsafe method - it modifies something. In your case you are only retrieving data.